### PR TITLE
[v8] Enable canned ACL for S3 (#9042)

### PR DESCRIPTION
**Pull request by atburke**
This change allows admins to specify a canned ACL when using S3. Backport of #9042.

### CODE REVIEW

**Review by russjones — APPROVED** [review]

File: docs/pages/setup/reference/backends.mdx [hunk 1]
```diff
@@ -209,13 +209,14 @@ running on an EC2 instance with an IAM role.
 
 These optional `GET` parameters control how Teleport interacts with an S3 endpoint, including S3-compatible endpoints.
 
-`s3://bucket/path?region=us-east-1&endpoint=mys3.example.com&insecure=false&disablesse=false`
+`s3://bucket/path?region=us-east-1&endpoint=mys3.example.com&insecure=false&disablesse=false&acl=private`
```
Comment (ptgott) [inline docs/pages/setup/reference/backends.mdx:212]: I think we can make it a bit more explicit that this URL is used to illustrate the arrangement of the params below. If you don't know the params are coming next, the role of the URL will take a bit of figuring out.

File: docs/pages/setup/reference/backends.mdx [hunk 2]
```diff
@@ -209,13 +209,14 @@ running on an EC2 instance with an IAM role.
...
 
 - `region=us-east-1` - set the Amazon region to use.
 - `endpoint=mys3.example.com` - connect to a custom S3 endpoint.
 - `insecure=true` - set to `true` or `false`. If `true`, TLS will be disabled.
 - `disablesse=true` - set to `true` or `false`. If `true`, S3 server-side encryption will be disabled. If `false`, aws:kms (Key Management Service) will be used for server-side encryption. Other SSE types are not supported at this time.
 - `sse_kms_key=kms_key_id` - If set to a valid AWS KMS CMK key ID all objects uploaded to S3 will be encrypted with this key. Details can be found below.
+- `acl=private` - set the [canned ACL](https://docs.aws.amazon.com/AmazonS3/latest/userguide/acl-overview.html#canned-acl) to use.
```
Comment (ptgott) [inline docs/pages/setup/reference/backends.mdx:219]: Should we specify that the value must be one of the values mentioned in the linked guide?

File: docs/pages/setup/reference/backends.mdx [hunk 3]
```diff
@@ -329,6 +330,47 @@ This is needed to detect session recordings.
 }
 ```
 
+### ACL Example: Transferring Object Ownership
```
Comment (ptgott) [inline docs/pages/setup/reference/backends.mdx:333]:
```suggestion
### ACL example: transferring object ownership
```

Let's make headings sentence case per the style guide (https://goteleport.com/docs/docs/best-practices/#style-guide)

File: docs/pages/setup/reference/backends.mdx [hunk 4]
```diff
@@ -329,6 +330,47 @@ This is needed to detect session recordings.
...
+
+If you are uploading from account `A` to a bucket owned by account `B` and want `A` to retain ownership of the objects, you can take one of two approaches.
```
Comment (ptgott) [inline docs/pages/setup/reference/backends.mdx:335]: Does "account" mean an AWS account?

File: docs/pages/setup/reference/backends.mdx [hunk 5]
```diff
@@ -329,6 +330,47 @@ This is needed to detect session recordings.
...
+
+#### Without ACL
```
Comment (ptgott) [inline docs/pages/setup/reference/backends.mdx:337]:
```suggestion
#### Without ACLs
```

File: docs/pages/setup/reference/backends.mdx [hunk 6]
```diff
@@ -329,6 +330,47 @@ This is needed to detect session recordings.
...
+
+If ACLs are disabled, object ownership will be set to `Bucket owner enforced` and no action is needed.
```
Comment (ptgott) [inline docs/pages/setup/reference/backends.mdx:339]:
```suggestion
If ACLs are disabled, object ownership will be set to `Bucket owner enforced` and no action will be needed.
```
for consistency

File: docs/pages/setup/reference/backends.mdx [hunk 7]
```diff
@@ -329,6 +330,47 @@ This is needed to detect session recordings.
...
+
+#### With ACL
```
Comment (ptgott) [inline docs/pages/setup/reference/backends.mdx:341]:
```suggestion
#### With ACLs
```

File: docs/pages/setup/reference/backends.mdx [hunk 8]
```diff
@@ -329,6 +330,47 @@ This is needed to detect session recordings.
...
+                    "s3:x-amz-acl": "bucket-owner-full-control"
+                }
+            }
+        }
+    ]
+}
+
+```
+
+For more information, see the [AWS Docs](https://docs.aws.amazon.com/AmazonS3/latest/userguide/about-object-ownership.html).
```
Comment (ptgott) [inline docs/pages/setup/reference/backends.mdx:372]:
```suggestion
For more information, see the [AWS documentation](https://docs.aws.amazon.com/AmazonS3/latest/userguide/about-object-ownership.html).
```

**Review by ptgott — APPROVED** [review]

**Review by timothyb89 — APPROVED** [review]

**Pull request merged**